### ISSUE TITLE
Decide on the encryption strategy for items/files

### ISSUE THREAD

**Issue by csuwildcat** [decentralized-identity/identity-hub#59]
Requirements to consider:

- Encryption should allow for external entities to access and decrypt the items they are granted permission for.
- Encryption should not require undue duplication and expensive client processes to effect (e.g. re-encryption/re-keying)
- Permissions can change dynamically, adding or removing external entities from access to future modifications of items. We need to decide how we will deal with this, which has impacts across the way encryption and items are structured/linked/bucketed at lower levels in the stack.

**Comment by codi0** [comment]
Are there any particular candidate protocols / implementations being looked at for encryption-based ACL? I'm thinking stuff like Peergos or WNFS.

If this has been addressed elsewhere, please point me in its direction. I'm trying to understand what cryptographic access control protocols exist, so I'm interested in where discussion has gotten on this one.

**Comment by Identitywoman** [comment]
@csuwildcat will reply

**Comment by andorsk** [comment]
@csuwildcat is this done?

**Comment by csuwildcat** [comment]
Yes, this is done, but let's close this when the work is complete and the spec text reflects it.

**Comment by andorsk** [comment]
@thehenrytsai will try to aim for next meeting.
